lib/player.ex

```diff
@@ -13,11 +55,14 @@ defmodule ExBeans.Player do
 #  @game_api Application.fetch_env!(:exbeans, :game_api)
 
   @typedoc "The player's reference."
-  @type playerName :: atom() | pid()
+  @type playerName   :: atom() | pid()
+
+  @type stateName    :: :no_turn | :initial_cards | :play_cards | :discard | :bonus_cards | :end_turn
+  @type notifyPlayer :: ((stateName(), Hand.hand, BeanField.beanField) -> :ok)
 
   defmodule State do
     @moduledoc false
-    defstruct name: nil, game: nil, hand: Hand.new(), field: BeanField.new(), score: 0, cards_played: 0
+    defstruct name: nil, game: nil, hand: Hand.new(), field: BeanField.new(), score: 0, cards_played: 0, notifyPlayer: nil
```

Comment:
Maybe default this to a fun that does nothing so it can be used without callbacks (e.g. for unit testing)